### PR TITLE
Change addon from Xendit to Xendit POS

### PR DESCRIPTION
**Pull request by andykim**
## Description
- Change addon from Xendit to Xendit POS
- Add filter company_id to get Xendit setting

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] General improvement (non-breaking feature that might not necessarily impact customer)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code (if necessary), particularly in hard-to-understand areas
- [ ] My changes generate no new **vulnerabilities**
- [ ] I have added tests that prove my fix is effective or that my feature works

## Merging
- [ ] I am aware which environment I intend to merge (staging | production)
- [ ] I am aware that a deployment will be triggered if I merge to staging and release branch
- [ ] I informed the team about my merge if it has risk of breaking the target API

## Deployment
- [ ] Unit tests passed
- [ ] UAT passed and communicated to QA and Product is informed about this deployment
- [ ] I'm aware off all the commits / merges behind that will be included in this deployment
- [ ] Feature flags are ready and communicated with the team

Please follow our [deployment guideline](https://xendit.atlassian.net/wiki/spaces/TPI/pages/2331476248/Engineer+Onboarding#Releasing-Flow).

## Screenshot

`put screenshot here if there are any ui changes`

### CODE REVIEW

**Review by candrasaputra — APPROVED** [review]

**Comment by candrasaputra** [comment]
thanks @andykim 🙏

**Review by rendegosling — COMMENTED** [review]
Some clarification and suggestions

File: xendit_pos/models/xendit_client.py [hunk 1]
```diff
@@ -13,16 +13,22 @@
 class XenditClient():
 
     tpi_server_domain = "https://tpi.xendit.co"
```
[the remark this answers is not in the document]
Comment (andykim) [reply xendit_pos/models/xendit_client.py]: @rendegosling I changed to `tpi-service-url`

File: xendit_pos/models/xendit_client.py [hunk 2]
```diff
@@ -13,16 +13,22 @@
...
+    company_id = ""
```
[the remark this answers is not in the document]
Comment (andykim) [reply xendit_pos/models/xendit_client.py]: @rendegosling It's the odoo company_id. I already changed it to `odoo_company_id` to more clearly

File: xendit_pos/models/xendit_client.py [hunk 3]
```diff
@@ -13,16 +13,22 @@
...
-                ('use_payment_terminal', '=', 'xendit')
+                ('use_payment_terminal', '=', 'xendit_pos'),
+                ('company_id', '=', self.company_id)
             ], limit=1)
-        return payment_method.xendit_secret_key
+        return payment_method.xendit_pos_secret_key
 
     def _get_xendit_test_mode():
-        payment_method = request.env['pos.payment.method'].sudo().search([('use_payment_terminal', '=', 'xendit')], limit=1)
+        payment_method = request.env['pos.payment.method'].sudo().search([('use_payment_terminal', '=', 'xendit_pos')], limit=1)
```
[the remark this answers is not in the document]
Comment (andykim) [reply xendit_pos/models/xendit_client.py]: @rendegosling I grouped them into 1 place. Thanks @rendegosling

File: xendit_pos/models/xendit_client.py [hunk 4]
```diff
@@ -13,16 +13,22 @@
...
+
+    def _get_xendit_api_key(self):
         payment_method = request.env['pos.payment.method'].sudo().search(
             [
-                ('use_payment_terminal', '=', 'xendit')
+                ('use_payment_terminal', '=', 'xendit_pos'),
+                ('company_id', '=', self.company_id)
             ], limit=1)
-        return payment_method.xendit_secret_key
+        return payment_method.xendit_pos_secret_key
```
[the remark this answers is not in the document]
Comment (andykim) [reply xendit_pos/models/xendit_client.py]: it's not encrypted yet when saved to db @rendegosling

File: xendit_pos/views/point_of_sale_assets.xml [hunk 5]
```diff
@@ -2,9 +2,9 @@
 <odoo>
     <template id="assets" inherit_id="point_of_sale.assets">
         <xpath expr="." position="inside">
-            <script type="text/javascript" src="/xendit/static/src/js/payment_xendit.js"></script>
-            <script type="text/javascript" src="/xendit/static/src/js/models.js"></script>
-            <script type="text/javascript" src="/xendit/static/src/js/xendit_qrcode_popup.js"></script>
+            <script type="text/javascript" src="/xendit_pos/static/src/js/payment_xendit_pos.js"></script>
+            <script type="text/javascript" src="/xendit_pos/static/src/js/models.js"></script>
+            <script type="text/javascript" src="/xendit_pos/static/src/js/xendit_qrcode_popup.js"></script>
```
Comment (rendegosling) [inline xendit_pos/views/point_of_sale_assets.xml:7]: I'm thinking that the name of the root folder should be `xendit_odoo_pos` for clarity. If we use `xendit_pos` it sounds generic to any POS. wdyt?

**Review by rendegosling — APPROVED** [review]
Looks generally OK. some comments that can be considered on a separate PR

File: xendit_pos/views/point_of_sale_assets.xml (hunk 5 above)
Comment (andykim) [reply xendit_pos/views/point_of_sale_assets.xml:7]: > Looks generally OK. some comments that can be considered on a separate PR

Thanks @rendegosling . I will update them on the next PR.

**Pull request merged**